src/plugins/lua/LuaThread.cpp

```diff
@@ -657,4 +658,80 @@ void LuaThread::AssociateTopLevelObject(const MQTopLevelObject* tlo)
 
 //============================================================================
 
+// TODO: Remove the debugging statements
+void LuaThread::InitializeSpawnTable()
+{
+	if (m_globalState["__spawns"] == sol::nil)
+	{
+		sol::table spawn_table = m_globalState.create_named_table("__spawns");
+
+		if (pSpawnManager != nullptr)
+		{
+			auto spawn = pSpawnManager->FirstSpawn;
+
+			while (spawn != nullptr)
+			{
+				WriteChatf("Adding spawn %s", spawn->Name);
+				spawn_table[spawn->SpawnID] = bindings::lua_MQTypeVar(datatypes::pSpawnType->MakeTypeVar(spawn));
+				spawn = spawn->GetNext();
+			}
+		}
+	}
+}
+
+void LuaThread::AddSpawn(eqlib::PlayerClient* spawn)
+{
+	if (m_globalState["__spawns"] != sol::nil)
+	{
+		WriteChatf("Adding spawn %s", spawn->Name);
+		m_globalState["__spawns"][spawn->SpawnID] = bindings::lua_MQTypeVar(datatypes::pSpawnType->MakeTypeVar(spawn));
+	}
```

Comment:
These functions are doing the `__spawns` lookup twice. We should save the result from the first so we can cut these down in half (they happen in bulk)

Reply:
good call -- I did that, but would it be better if I stored these tables as members on LuaThread directly? I'd have to update it when it gets set from nil, but that should work as well, no?

Reply:
yeah that would be even better if it works as expected.

Reply:
I added and tested this, with an extra guard against trying to write/access lua tables that don't exist anymore due to a finished or bad lua state